### DISCUSSION TITLE
Save/Load Networks

### DISCUSSION THREAD

**Discussion by latot** [luukvdmeer/sfnetworks#264] [Ideas]
Hi all, I think more than one ppl would like something like this, that would be great be able to save and load networks!

I know there is no... standard about this, but I think even a simpler solution based on actual tech could also be very useful.

My idea would be something like save a sfnetwork object in a gpkg where every table would be:

- nodes: save nodes here
- edges: save edges here
- info: save things like of is directed or not, related info that is need about the network

That is the base, personally this could helps a lot!

Just in the side, because before this, I think would be useful the format be little closer to this:
https://www.gaia-gis.it/fossil/libspatialite/wiki?name=topo-layout

Is a lot easier to work and handle networks when there is a column with an ID, because with the actual way, any change on a network will change the nodes/edges assignation.

Personally, I found useful the faces data, but is a little different on networks, because not all of them belongs to a face.

Thx

**Comment by luukvdmeer** [comment]
I agree this would be useful, and not too hard if we stick to a simple format such as GPKG with two layers (nodes and edges). Anyone knows if GPKG has some kind of metadata where it is possible to store the info about directionality etc? Or it might be better either way to specify yourself if you want to read the networks as directed or not: `read_sfnetwork("foo/bar.gpkg", directed = TRUE)`

**Comment by latot** [comment]
technically, gpkg is a sqlite format, so you can create, edit or use any table to keep the metadata.

I think the best is specify if is directed or not in the metadata, the reason is just all networks are from one or the other type, I mean in how they are constructed, would be better to keep that on the track and clear. About that would be even better be able to save some info about the network, like what it is, for what is used, if is directed, all this could be saved inside the gpkg!

Also, would be good to keep the version of sfnetworks on it, because of issues like https://github.com/luukvdmeer/sfnetworks/issues/259, then if the file need to be updated there can be message for example.

Also, a middle point, would be store metadata in a table as JSON format, then read from R.

**Comment by agila5** [comment]
Just for future reference (and please notice that I didn't check the details), the approach taken by `OSMnx` is to save the data using `GeoPackage`/`XML`/`Graphml` formats: https://osmnx.readthedocs.io/en/stable/user-reference.html#module-osmnx.io
